app/controllers/api/providers_controller.rb

```diff
@@ -51,13 +51,11 @@ def refresh_resource(type, id = nil, _data = nil)
     end
 
     def delete_resource(type, id = nil, _data = nil)
-      raise BadRequestError, "Must specify an id for deleting a #{type} resource" unless id
-
-      api_action(type, id) do |klass|
-        provider = resource_search(id, type, klass)
-        api_log_info("Deleting #{provider_ident(provider)}")
-
-        destroy_provider(provider)
+      delete_action_handler do
+        raise BadRequestError, "Must specify an id for deleting a #{type} resource" unless id
+        provider = resource_search(id, type, collection_class(type))
+        task = provider.class.schedule_destroy_queue(provider.id)
+        action_result(true, "#{provider_ident(provider)} deleting", :task_id => task.id)
```

Comment:
I think you should use `destroy_queue` instead of `schedule_destroy_queue`  because it also schedules the `destroy` of child managers